### PR TITLE
Initial Execution and Interpreter docs

### PR DESCRIPTION
**Pull request by carlitosan**
Initial Execution and Interpreter docs

### CODE REVIEW

File: content/docs/user-guide/scripting/script-canvas/editor-reference/interpreter.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+linktitle: Interpreter
+title: Script Canvas Interpreter
+description: Learn how to use the Script Canvas Interpreter in embedded in the Script Canvas Editor in the Open 3D Engine (O3DE).
```
Comment (cgalvan) [inline content/docs/user-guide/scripting/script-canvas/editor-reference/interpreter.md]: Should `in embedded` just be `embedded` ?

File: content/docs/user-guide/scripting/script-canvas/editor-reference/interpreter.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+weight: 100
+---
+
+Script Canvas offers an interpreter interface that can be placed almost in almost any situation in the engine or the editor. The Script Canvas Editor itself makes use of it in the tools window.
```
Comment (cgalvan) [inline content/docs/user-guide/scripting/script-canvas/editor-reference/interpreter.md]: `almost in almost` -> `in almost`

File: content/docs/user-guide/scripting/script-canvas/programmer-guide/execution.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+
+![Script Canvas Execution Classes Diagram](/images/user-guide/scripting/script-canvas/script-canvas-execution-classes-uml.png)
+
+## ExecutionState ##
+`ExecutionState` is the abstract class which provides an interface for the actual runtime execution of compiled Script Canvas graphs. A compiled graph asset will store the traits which detemine whichchild class of `ExecutionState` will be required to execute it. Wether or not the graph defines pure functionality (it only defines methods that operate solely on their inputs), or defines state that will last beyond the execution of the initial function is read by `ExecutionStateStorage` and the appropriate child class of `ExecutionState` is created.
```
Comment (cgalvan) [inline content/docs/user-guide/scripting/script-canvas/programmer-guide/execution.md]: `detemine whichchild` -> `determine which child`

Comment (cgalvan) [reply content/docs/user-guide/scripting/script-canvas/programmer-guide/execution.md]: `Wether` -> `Whether`

File: content/docs/user-guide/scripting/script-canvas/programmer-guide/execution.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+Since `ExecutionState` expects to run on a good runtime data provided to it with no overhead, it performs no safety checks against bad data. As an abstract class bordering on a pure interface, the `ExecutionState` object refers to, but owns as little as possible, any input it requires from the user to begin and maintain execution.
+
+Developers who wish to embed Script Canvas runtime functionality somewhere in the engine where it currently does not exist only need to to use a properly configured `ExecutionState` object to do so. They can accomplish this however they wish, but O3DE provides some convienence classes (with performance in mind) to make the job easier. All of these convenience classes are used by the Script Canvas `RuntimeComponent` and the Script Canvas `Interpreter`. The `RuntimeComponent` is an example of the execution running after very strict safey checks for proper runtime data being provided. The `Interpreter` is an example of active, real-time, safety checking for valid runtime data, which maybe under rapid iteration by a Script Canvas author.
```
Comment (cgalvan) [inline content/docs/user-guide/scripting/script-canvas/programmer-guide/execution.md:20]: `safey` -> `safety`

File: content/docs/user-guide/scripting/script-canvas/programmer-guide/execution.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+What follows are descriptions of the convenience classes for embedding an `ExecutionState`.
+
+## ExecutionStateStorage ##
+`ExecutionStateStorage` provides enough static storage space to allow the construction of any child class of `ExecutionState` without additional heap allocation. It is designed to be used with `ExecutionStateHandler`.
+
+## ExecutionStateHandler ##
+`ExecutionStateHandler` provides RAII semantics for the `ExecutionStateStorage` and the `ExecutionState` owned by that storage. It also provides a measure of error reporting for the violation of safety checks if the developer embeds it in O3DE in such a way that it attempts to execute ScriptCanvas runtime data that is not valid. In non-release builds, this is done enabled with the preprocessor macro, `SC_RUNTIME_CHECKS_ENABLED`. This macros turns on basic, minimal error checking and early returns, while minimizing branching at runtime. If the macro is disabled, the errors are written as asserts (if asserts are enabled in the build) with zero branching.
```
Comment (cgalvan) [inline content/docs/user-guide/scripting/script-canvas/programmer-guide/execution.md]: `done enabled` -> either just `done` or `enabled`

**Review by cgalvan — CHANGES_REQUESTED** [review]
Just some minor typos, other than that looks good!

Also this PR should be targeted at the `o3de:main` instead of your fork's `main` when you are ready.

File: content/docs/user-guide/scripting/script-canvas/editor-reference/interpreter.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+![Script Canvas Interpreter Widget](/images/user-guide/scripting/script-canvas/script-canvas-interpreter-widget.png)
+
+The Interpreter take a Scripts Canvas source file and allows the user to configure and execute it immediately. If the source file produces more than a single thread of execution that starts and stops immediately after "On Graph Start", the script will continue to execute indefinitely. The user can press the stop button or close the window to stop the widget from executing.
+
+![Script Canvas Interpreter Widget](/images/user-guide/scripting/script-canvas/script-canvas-interpreter-widget-opened.png)
+
+Note that many of the nodes, and much of the functionality available for Entities during the simulation, will not apply when the Interpreter executes. When references to 'Self' are detected, the interpreter will not run the graph.
```
Comment (onecent1101) [inline content/docs/user-guide/scripting/script-canvas/editor-reference/interpreter.md:17]: Minor: you can wrap the content inside Note
```
{{< note >}}
...
{{< /note >}}
```

**Pull request merged**